Source/WebKit/WebProcess/GPU/GPUProcessConnection.h

```diff
@@ -101,6 +101,8 @@ class GPUProcessConnection : public ThreadSafeRefCountedAndCanMakeThreadSafeWeak
     void setMediaEnvironment(WebCore::PageIdentifier, const String&);
 #endif
 
+    void notifyMessageDidTimeout();
```

Comment:
Not sure this "message did timeout" is the right concept here. It's not really related to all messages.
Maybe terminateProcess() would sound better?

I'm not very keen on the flagging either, eg..the m_hasTimeout. I'd very much expect that the connection object would go away directly at the call, similar to after connection closing. I do understand it seems to be easier to implement that you'd do something that closes the connection, and then let the connection events handle the actual teardown..

Reply:
At some point we need to switch from the symptom (message timed out), to the requested action (terminate the GPU process). I don't have strong opinions on where that boundary is really, happy to make it not-here if you prefer.

I'm worried that making the connection object go away immediately adds a new tear down path that isn't used anywhere else. We have existing code for making sure we recreate everything in response to a crash, so making use of that seems beneficial.

Reply:
Yeah, I think the symptom doesn't belong here anymore..